rest_docs/rest-catalog-open-api-v0.1.yaml

```diff
@@ -0,0 +1,657 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+---
+openapi: 3.0.3
+info:
+  title: Apache Iceberg REST Catalog API
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+  version: 1.0.0
+  description:
+    Defines the specification for the first version of the REST Catalog API. Implementations should support both Iceberg table specs v1 and v2, with priority given to v2.
+servers:
+  - url: https://{host}:{port}/{basePath}
```

Comment:
Not sure whether it's useful to have a server/url definition in the spec for Iceberg. This particular one makes it mandatory to use 3 variables, which makes it complicated to "just use" the correct URL provided by the (web/REST) server implementation.

Reply:
It wouldn't be very useful to use the Swagger UI to make requests if we couldn't define the host etc.

I don't know how one would use the swagger UI to make calls without having a server variable, given that people's host will surely be different.

It was @rymurr who asked me to make it configurable in a comment that is.... somewhere deep within all of the comments of this PR.